### PR TITLE
Simple clarifications

### PR DESCRIPTION
**Pull request by gunney1**
Improve code docs with some comments and a Doxygen set-up.  Changed file names to match class names.  Changed some class names better describe the class's role.

### CODE REVIEW

File: src/core/imports/mpi.cpp [hunk 1]
```diff
@@ -1831,7 +1831,8 @@ std::vector<T> AllToAll(
     Comm const& comm )
 EL_NO_RELEASE_EXCEPT
 {
-    LogicError("AllToAll: Is this used? Tell Tom if so.");
+    // LogicError("AllToAll: Is this used? Tell Tom if so.");
+    /* Tom: this is used when the Dot test is run with 2 processes.  Debian OS with mpich. --BTNG */
```
Comment (timmoon10) [inline src/core/imports/mpi.cpp:1835]: Let's open an issue, even if we're not planning on fixing this anytime soon.

File: tests/blas_like/Axpy.cpp [hunk 2]
```diff
@@ -56,14 +59,17 @@ void TestCorrectness
...
+            Print(E, "E (error)");
         Output
             ("|| E ||_F / || Y ||_F = ",
              EFrobNorm, "/", YFrobNorm, "=", EFrobNorm/YFrobNorm);
     }
 }
 
 
+/*!
+  @brief Exercise Ax+y and check or correctness.
```
Comment (timmoon10) [inline tests/blas_like/Axpy.cpp]:
```suggestion
  @brief Exercise Ax+y and check for correctness.
```

File: include/El/core/DistMatrix/ElementMatrix/MC_MR.hpp [hunk 3]
```diff
@@ -293,6 +293,7 @@ class DistMatrix<Ring,MC,MR,ELEMENT,Dev>
 
     void do_empty_data_() override;
 
+    // Make friend of *all* DistMatrix classes(?)
```
Comment (timmoon10) [inline include/El/core/DistMatrix/ElementMatrix/MC_MR.hpp]:
```suggestion
    // Make friend of *all* DistMatrix classes
```

**Review by timmoon10 — APPROVED** [review]
LGTM.

**Review by benson31 — APPROVED** [review]
LGTM. @gunney1 Please commit the suggestions that @timmoon10  recommended to some of the comments. It told me I didn't have permission to do it. Once that's done, I'll merge.

**Pull request merged**